### ISSUE TITLE
Indentation error for indented comments

### ISSUE THREAD

**Issue by Conaclos** [eslint/eslint#8987]
**Tell us about your environment**

* **ESLint Version:** 4.3.0
* **Node Version:** 8.2.1
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    ...
    indent: [2, 4]
}
```
</details>

**What did you do? Please include the actual source code causing the issue.**

```js
let a = 3
    // Assign 3 to a
```

**What did you expect to happen?** no indentation error.

**What actually happened? Please include the actual, raw output from ESLint.**

```
2:1 error  Expected indentation of 0 spaces but found 4  indent
```

**Comment by platinumazure** [comment]
This is working as designed- we enforce comment indentation as of ESLint 4.x. You may use the `indent-legacy` rule if you want the old behavior.

In the new rule, we support comments being aligned with the start of the previous line, the start of the next line, or the same indent level as where a line of code "should be" if code were present at the comment's location.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.

**Comment by benpptung** [comment]
Can we ignore hierarchy comment ?
For example,

```
/*
 * error handling
 */

  // catch 404
app.use(....)
....

  // final handler
app.use(...)
....

``` 

I'm asking because if the project is big, and complicated, I need hierarchy comments ..
Currently, eslint indent is good, however, it's bit of frustrated, no hierarchy comments support.
It forces me to put all comments with same ident.

**Comment by not-an-aardvark** [comment]
@benpptung I'm not quite sure what you mean by "hierarchy comments", but feel free to create a new issue if you have a proposal. Keep in mind that if you want ESLint to use special behavior for hierarchy comments, ESLint will need to be able to precisely determine whether a given comment is a hierarchy comment.
